scripts/imitation_learning/isaaclab_mimic/consolidated_demo.py

```diff
@@ -0,0 +1,485 @@
+# Copyright 2025 Enactic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Script to record teleoperated demos and run mimic dataset generation in real-time.
+"""
+
+# Launching Isaac Sim Simulator first.
+
+import argparse
+
+from isaaclab.app import AppLauncher
+
+# add argparse arguments
+parser = argparse.ArgumentParser(
+    description="Record demonstrations and run mimic dataset generation for Isaac Lab environments."
+)
+parser.add_argument("--task", type=str, default=None, help="Name of the task.")
+parser.add_argument(
+    "--num_demos", type=int, default=0, help="Number of demonstrations to record. Set to 0 for infinite."
+)
+parser.add_argument(
+    "--num_success_steps",
+    type=int,
+    default=10,
+    help="Number of continuous steps with task success for concluding a demo as successful. Default is 10.",
+)
+parser.add_argument(
+    "--num_envs",
+    type=int,
+    default=5,
+    help=(
+        "Number of environments to instantiate to test recording and generating datasets. The environment specified by"
+        " `teleop_env_index` will be used for teleoperation and recording while the remaining environments will be used"
+        " for real-time data generation. Default is 5."
+    ),
+)
+parser.add_argument(
+    "--teleop_env_index",
+    type=int,
+    default=0,
+    help="Index of the environment to be used for teleoperation. Set -1 for disabling the teleop robot. Default is 0.",
+)
+parser.add_argument("--teleop_device", type=str, default="keyboard", help="Device for interacting with environment.")
+parser.add_argument(
+    "--step_hz", type=int, default=0, help="Environment stepping rate in Hz. Set to 0 for maximum speed."
+)
+parser.add_argument("--input_file", type=str, default=None, help="File path to the source demo dataset file.")
+parser.add_argument(
+    "--output_file",
+    type=str,
+    default="./datasets/output_dataset.hdf5",
+    help="File path to export recorded episodes.",
+)
+parser.add_argument(
+    "--generated_output_file",
+    type=str,
+    default=None,
+    help="File path to export generated episodes by mimic.",
+)
+# append AppLauncher cli args
+AppLauncher.add_app_launcher_args(parser)
+# parse the arguments
+args_cli = parser.parse_args()
+
+# launch the simulator
+app_launcher = AppLauncher(args_cli)
+simulation_app = app_launcher.app
```

Comment:
How about doing them in `if __name__ == "__main__":` or `main()`?
Or how about removing `if __name__ == "__main__":`?
If we have runnable code outside of  `if __name__ == "__main__":`, we can't use this as a library.